lib/impure/nre.nim

```diff
@@ -198,7 +198,7 @@ type
   Captures* = distinct RegexMatch
   CaptureBounds* = distinct RegexMatch
 
-  RegexError* = ref object of Exception
+  RegexError* = ref object of CatchableError
 
   RegexInternalError* = ref object of RegexError
```

Comment:
I debated making RegexInternalError extend Defect instead of RegexError but didn't do it, should I do it?